nmsg/msgmod/transparent_message.c

```diff
@@ -78,6 +78,45 @@ nmsg_message_get_field_idx(struct nmsg_message *msg, const char *field_name,
 }
 
 nmsg_res
+nmsg_message_get_num_field_values_by_idx(struct nmsg_message *msg,
+					 unsigned field_idx,
+					 size_t *n_field_values)
+{
+	struct nmsg_msgmod_field *field;
+
+	CHECK_TRANSPARENT();
+	GET_FIELD(field_idx);
+	DESERIALIZE();
+
```

Comment:
At this point we need to check if the field is a virtual field, otherwise the `field->descr->...` dereference in the following lines will dereference a NULL pointer.

Something like...

```c
        if (field->descr == NULL) {
                if (field->get == NULL) {
                        return (nmsg_res_failure);
                }
                *n_field_values = 0;
                nmsg_res res = nmsg_res_success;
                for (unsigned val_idx = 0; res == nmsg_res_success; val_idx++) {
                        void *data = NULL;
                        size_t len = 0;

                        res = field->get(msg, field, val_idx, &data, &len, msg->msg_clos);
                        if (res == nmsg_res_success) {
                                *n_field_values += 1;
                        }
                }
                return (nmsg_res_success);
        }
```